### PR TITLE
Enable boolean type test on GPU.

### PR DESCRIPTION
**Pull request by bader**
https://github.com/intel/llvm/pull/2458 fixes issue 2264.

### CODE REVIEW

**Comment by bader** [comment]
https://github.com/intel/llvm/pull/2458 is merged.
@vladimirlaz, do I understand it correctly that I should wait for CI to pick up the first night build with https://github.com/intel/llvm/pull/2458 to be ready before merging this patch to avoid false alarms?

**Comment by vladimirlaz** [comment]
> [intel/llvm#2458](https://github.com/intel/llvm/pull/2458) is merged.
> @vladimirlaz, do I understand it correctly that I should wait for CI to pick up the first night build with [intel/llvm#2458](https://github.com/intel/llvm/pull/2458) to be ready before merging this patch to avoid false alarms?

You are right. The nightly happens around 7-8PM MSK.

**Comment by bader** [comment]
@llvm-reviewers-runtime, @againull, ping.

**Review by againull — APPROVED** [review]

**Pull request merged**
